File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void swap(vector<int>& arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    void sortColors(vector<int>& nums) {
        int left = 0;
        int mid = 0;
        int right = nums.size() - 1;
        while(mid <= right) {
            if(nums[mid] == 0) {
                swap(nums, mid++, left++);
            }
            else if(nums[mid] == 2) {
                swap(nums, mid, right--);
            }
            else {
                swap(nums, mid++, left);
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    void swap(int[] arr, int a, int b) {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }
    public void sortColors(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        while(left < right) {
            if(nums[right] == 2) {
                right--;
                continue;
            }
            else if(nums[left] == 0) {
                left++;
                continue;
            }
            else {
                swap(nums, left, right);
                if(nums[right] == 2) right--;
                if(nums[left] != 2) left++;
            }
        }
        left = 0;
        while(left < right) {
            if(nums[right] == 2) {
                right--;
                continue;
            }
            else if(nums[left] == 0) {
                left++;
                continue;
            }
            else {
                swap(nums, left, right);
                if(nums[right] != 0) right--;
                if(nums[left] == 0) left++;
            }
        }
    }
}